AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -7,7 +7,13 @@
 
 import Foundation
 
-class EndpointClient: InternalPinpointClient {    
+class EndpointClient: InternalPinpointClient {
+    internal var context: PinpointContext
```

Comment:
`context` should be `unowned` in **all** classes that conform to `InternalPinpointClient` to avoid a retain cycle, as the context has strong references to all clients. 

These relationships mirror what the Pinpoint SDK had, perhaps once we have all of them properly implemented we can check if we really do need to have a reference to `context` in each client, or if we just can pass the required values instead.

But for now, they need to be `unowned`

Reply:
Also, do we need the `internal`  to be explicit?

Reply:
> Also, do we need the internal to be explicit? 
Nope, thanks for catching that!

> as the context has strong references to all clients.
yikes, adding `unowned` modifier for now